server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java

```diff
@@ -483,55 +479,13 @@ protected Rule toDoc(Map<String, Object> fields) {
     return tags;
   }
 
-  /**
-   * @deprecated please use getByKey(RuleKey key)
-   */
-  @Deprecated
-  @CheckForNull
-  public Rule getById(int id) {
-    SearchRequestBuilder request = getClient().prepareSearch(this.getIndexName())
-      .setTypes(this.getIndexType())
-      .setQuery(QueryBuilders.termQuery(RuleNormalizer.RuleField.ID.field(), id))
-      .setSize(1);
-    SearchResponse response = request.get();
-
-    SearchHit hit = response.getHits().getAt(0);
-    if (hit == null) {
-      return null;
-    } else {
-      return toDoc(hit.getSource());
-    }
-  }
+  private enum ToRuleKey implements Function<String, RuleKey> {
+    INSTANCE;
 
-  /**
-   * @deprecated please use getByKey(RuleKey key)
-   */
-  @Deprecated
-  public List<Rule> getByIds(Collection<Integer> ids) {
-    SearchRequestBuilder request = getClient().prepareSearch(this.getIndexName())
-      .setTypes(this.getIndexType())
-      .setSearchType(SearchType.SCAN)
-      .setScroll(TimeValue.timeValueSeconds(3L))
-      .setSize(100)
-      .setQuery(QueryBuilders.termsQuery(RuleNormalizer.RuleField.ID.field(), ids));
-    SearchResponse scrollResp = request.get();
-
-    List<Rule> rules = newArrayList();
-    while (true) {
-      SearchScrollRequestBuilder scrollRequest = getClient()
-        .prepareSearchScroll(scrollResp.getScrollId())
-        .setScroll(TimeValue.timeValueSeconds(3L));
-
-      scrollResp = scrollRequest.get();
-
-      for (SearchHit hit : scrollResp.getHits()) {
-        rules.add(toDoc(hit.getSource()));
-      }
-      // Break condition: No hits are returned
-      if (scrollResp.getHits().getHits().length == 0) {
-        break;
-      }
+    @Override
+    public RuleKey apply(@Nonnull String input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)